index.js

```diff
@@ -97,11 +97,16 @@ Pool.prototype._create = function (cb) {
     if (err) {
       this.log('client connection error:', err)
       cb(err, null)
-    } else {
-      this.log('client connected')
-      this.emit('connect', client)
-      cb(null, client)
+      return
     }
+
+    this.log('client connected')
+    this.emit('connect', client)
+      .then(() => {
+        this.log('came back from connect')
+        cb(null, client)
+      })
+      .catch(err => cb(err, null))
```

Comment:
please add `this.log('connect listener error:', err)` before calling `cb`

Reply:
added